### ISSUE TITLE
healthcheck for config-server not working

### ISSUE THREAD

**Issue by dyroberts** [pivotal-cf/spring-cloud-services-starters#178]
Client app (v3.2.0.RELEASE) with:
```
management:
  endpoint:
    health:
      show-components: always
```
results in the following result from `/actuator/health`

```
{"status":"UP","components":{"clientConfigServer":{"status":"UNKNOWN","details":{"error":"no property sources located"}}, ...
```

**Comment by nniesen** [comment]
The attached app demonstrates the issue producing:
```
{"status":"UP","components":{"clientConfigServer":{"status":"UNKNOWN","details":{"error":"no property sources located"}},"discoveryComposite":{"description":"Discovery Client not initialized","status":"UNKNOWN","components":{"discoveryClient":{"description":"Discovery Client not initialized","status":"UNKNOWN"}}},"diskSpace":{"status":"UP","details":{"total":254506831872,"free":25237053440,"threshold":10485760,"exists":true}},"mongo":{"status":"UP","details":{"version":"4.0.6"}},"ping":{"status":"UP"},"refreshScope":{"status":"UP"}}}
```
[spring5-cfenv.zip](https://github.com/pivotal-cf/spring-cloud-services-starters/files/5906033/spring5-cfenv.zip)

**Comment by dyroberts** [comment]
Open-source spring config-client health-check is not working with the legacy bootstrap behaviour. To fix this issue I'll be doing the full migration to the new boot 2.4 config server feature. This is non-trivial as the scs auto-config relies heavily on bootstrap lifecycle.

**Comment by nniesen** [comment]
Is the "plan" to fix this before releasing, release with the known issue, or don't know yet?

**Comment by dyroberts** [comment]
@nniesen v3.2.0.RELEASE is available in maven central. It includes the known issue.
I'm working on the fix, which is to use the new spring-boot config.import feature. That will be v3.2.1.RELEASE which should be out early next week.

**Comment by dyroberts** [comment]
This is presently blocked by a config-client bug
https://github.com/spring-cloud/spring-cloud-config/issues/1803

**Comment by nniesen** [comment]
Unfortunately the 3.2.1.RELEASE breaks our startup configuration. I haven't had time to triage yet and maybe there's some upgrade change I need to make but with 3.2.0 and earlier, the first thing the app did was reach out to config server:

```
 :: Spring Boot ::                (v2.4.3)

2021-03-09 15:14:33.827  INFO 51856 --- [           main] c.c.c.ConfigServicePropertySourceLocator : Fetching config from server at : http://localhost:8888
```

With the 3.2.1.RELEASE that doesn't happen so beans fail to wire due to missing property placeholder values.
[scs-3.2.1-startup-error.txt](https://github.com/pivotal-cf/spring-cloud-services-starters/files/6111493/scs-3.2.1-startup-error.txt)

**Comment by dyroberts** [comment]
@nniesen if you could attach a sample app exposing the issue it would be appreciated. My sample is working with 3.2.1.RELEASE, other than the known-issue/bug mentioned above.

**Comment by nniesen** [comment]
Our actual usage is a bit more complicated but it boils down to @Value annotations not being resolved by the Property Placeholder from the config server when the application starts up because it no longer asks for config on startup.

```
@SpringBootApplication
public class DemoApplication {
    public static void main(final String[] args) {
        SpringApplication.run(DemoApplication.class, args);
    }

    @Value("${myapp.config.server.value}")
    private String value;
}
```
And and a config server with the property value:
`spring.cloud.config.server.overrides.myapp.config.server.value: gotIt`

[spring5-cfenv.zip](https://github.com/pivotal-cf/spring-cloud-services-starters/files/6125940/spring5-cfenv.zip)

**Comment by nniesen** [comment]
I figured out it's related to the Cloud 2020 Bootstrap change. The app starts fine with bootstrap enabled however that also causes the heathcheck to report unknown again. 😆

https://github.com/spring-cloud/spring-cloud-release/wiki/Spring-Cloud-2020.0-Release-Notes
> Bootstrap, provided by spring-cloud-commons, is no longer enabled by default. If your project requires it, it can be re-enabled by properties or by a new starter.
> 
> * To re-enable by properties set spring.cloud.bootstrap.enabled=true or spring.config.use-legacy-processing=true. These need to be set as an environment variable, java system property or a command line argument.
> 
> * The other option is to include the new spring-cloud-starter-bootstrap.

**Comment by dyroberts** [comment]
@nniesen I see your issue. You've got your `spring.application.name` property set in a `bootstrap.yml`. That should now be set in `application.yml`. `bootstrap.yml` is no longer required with spring-boot 2.4.

**Comment by dyroberts** [comment]
> I figured out it's related to the Cloud 2020 Bootstrap change. The app starts fine with bootstrap enabled however that also causes the heathcheck to report unknown again. 

The healthcheck issue is in spring-cloud 2020.0.1 and only affects legacy bootstrap behaviour. It is fixed in latest spring-cloud snapshots; but we've moved on from legacy bootstrap here, so we won't need to worry about it anyway.

**Comment by dyroberts** [comment]
Fixed in [v3.2.1.RELEASE](https://github.com/pivotal-cf/spring-cloud-services-starters/releases/tag/v3.2.1.RELEASE)
